src/storage/plugins/CurlStorage.cxx

```diff
@@ -493,24 +516,31 @@ class HttpListDirectoryOperation final : public PropfindOperation {
 	 */
 	gcc_pure
 	StringView HrefToEscapedName(const char *href) const noexcept {
+		StringView relative_path;
 		StringView path = uri_get_path(href);
 		if (path == nullptr)
 			return nullptr;
 
 		/* kludge: ignoring case in this comparison to avoid
 		   false negatives if the web server uses a different
 		   case */
-		path = StringAfterPrefixIgnoreCase(path, base_path.c_str());
-		if (path == nullptr || path.empty())
+		relative_path = StringAfterPrefixIgnoreCase(path, base_path.c_str());
+		if (relative_path == nullptr || relative_path.empty()) {
```

Comment:
at first I wanted to use `uri_is_relative` on `path` but I didn't find how to convert a StringView to a c_str as it doesn't have a c_str method...

Reply:
Maybe we should migrate `uri_is_relative()` to take a `StringView`?
`StringView` / `std::string_view` are not null-terminated, so there cannot be a `.c_str()` method.
A kludge for now could be to cast to `std::string` and call its `.c_str()` - which would duplicate the string.

Reply:
I tried with `uri_is_relative` but it wouldn't work because those Webdav hrefs do start with "/".

So I migrated and used `uri_has_scheme` instead to decide what base path to check against.
See 74b2fc7fdca9be13cbbe4cb52b2fab573b3cf82c

Reply:
Of course... if it starts with a slash, it's not relative, it's an absolute URI path.